addons/sale/models/account_move_line.py

```diff
@@ -155,12 +155,15 @@ def _sale_determine_order(self):
         for move_line in self:
             if move_line.analytic_distribution:
                 distribution_json = move_line.analytic_distribution
-                sale_order = self.env['sale.order'].search([('analytic_account_id', 'in', list(int(account_id) for account_id in distribution_json.keys())),
+                account_ids = [int(x) for x in ','.join(list(distribution_json.keys())).split(',')]
```

Comment:
```suggestion
                account_ids = [int(account_id) for key in distribution_json.keys() for account_id in key.split(',')]
```
Don't you think it's easier to read/understand like this (not tested)?